common/core/gpudevice.cpp

```diff
@@ -95,7 +95,7 @@ void GpuDevice::GetConnectedPhysicalDisplays(
   }
 }
 
-std::vector<NativeDisplay *> GpuDevice::GetAllDisplays() {
+std::vector<NativeDisplay *> &GpuDevice::GetAllDisplays() {
```

Comment:
make this const std::vector<NativeDisplay *> &, so no one accidentally manipulates it.